src/student/student.service.ts

```diff
@@ -134,200 +135,164 @@ export class StudentService {
+                url: JSON.stringify(portfolioUrl),
+                type: projectTypeEnum.portfolio,
+                updatedAt: dataUpdatedAt,
+            }
 
-    const setBonusProjectUrl: UpdateProjectUrlDto = {
-      url: JSON.stringify(bonusProjectUrl),
-      type: projectTypeEnum.bonusProject,
-      updatedAt: dataUpdatedAt,
-    };
-    const setPortfolioUrl: UpdateProjectUrlDto = {
-      url: JSON.stringify(portfolioUrl),
-      type: projectTypeEnum.portfolio,
-      updatedAt: dataUpdatedAt,
-    };
+            const githubValidator = await this.githubService.validateGithubName(student.id, githubName)
 
-    if (student) {
-      const userId = await this.getUserByStudentId(studentId);
-      const uniqueUserMail = await this.userRepository.find({
-        where: { email, id: Not(userId) },
-      });
-      const githubValidator = await this.githubService.validateGithubName(
-        studentId,
-        githubName,
-      );
+            const update1 = (!Object.values({...restOfDetails, githubName}).every(detail => detail === undefined)) || githubName
+                ?
+                await this.studentRepository.createQueryBuilder()
+                    .update('students')
+                    .set({
+                        githubName,
+                        ...restOfDetails,
+                        updatedAt: dataUpdatedAt,
+                    })
+                    .where('id = :id', {id: student.id})
+                    .execute()
+                :
+                console.log('no [students] rows affected')
+            //console.log(update1)
 
-      if (!(email === undefined) && uniqueUserMail.length > 0) {
-        throw new NotAcceptableException(
-          `Użytkownik z adresem e-mail: ${email} już istnieje w bazie`,
-        );
-      }
+            const update2 = (email)
+                ?
+                await this.userRepository.createQueryBuilder()
+                    .update('users')
+                    .set({
+                        email,   //updatedAt: () => 'CURRENT_TIMESTAMP',
+                    })
+                    .where('id = :id', {id: student.userId})
+                    .execute()
+                :
+                console.log('no [users] rows affected');
+            //console.log(update2)
 
-      const update1 =
-        !Object.values({ ...restOfDetails, githubName }).every(
-          (detail) => detail === undefined,
-        ) || githubName
-          ? await this.studentRepository
-              .createQueryBuilder()
-              .update('students')
-              .set({
-                githubName,
-                ...restOfDetails,
-                updatedAt: dataUpdatedAt,
-              })
-              .where('id = :studentId', { studentId })
-              .execute()
-          : console.log('no [students] rows affected');
-      //console.log(update1)
+            const urlsUpdate = this.projectRepository.createQueryBuilder()
```

Comment:
Tutaj mamy podobną sytuację.
Sama funkcja update'u powinna być napisana w project.service.ts a tutaj powinno być odniesienie do tego serwisu a nie bezpośrednio do repozytorium. Tak będzie wydaje mi się zgodnie z założeniami nestjs.

Reply:
to też uwzględnione :)